### PR TITLE
Feature/add unit test

### LINKED ISSUE #5: Add unit tests
Add some tests for this package.

### PR DESCRIPTION
**Pull request by midyan**
Closes #5 

Changes
------
- [x] Add test files
- [x] Add mocha as test driver
- [x] Add TravisCI integration
- [x] Mock HTTP requests to isolate test environment
# WIP

### CODE REVIEW

**Comment by mauriciovieira** [comment]
https://travis-ci.org/apinf/apinf-fiware/ is set

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+sudo: required
+language: node_js
+node_js:
+  - "8"
+
+before_install:
+  - "curl -L http://git.io/ejPSng | /bin/sh"
+
+env:
+  - CXX=g++-4.8
```
Comment (mauriciovieira) [inline .travis.yml:10]: It seems these two env lines run a pipeline each. https://travis-ci.org/apinf/apinf-fiware/builds/267003602?utm_source=github_status&utm_medium=notification We don't need travis to run twice.

File: fiware_server.js [hunk 2]
```diff
@@ -106,27 +105,38 @@ OAuth.registerService('fiware', 2, null, function(query) {
  * @return  {Object}              The response from the token request (see above)
  */
 const getTokens = function(config, query) {
-
+  // Endpoint for requestin access token
```
Comment (mauriciovieira) [inline fiware_server.js]: s/requestin/requesting/

File: fiware_server.js [hunk 3]
```diff
@@ -106,27 +105,38 @@ OAuth.registerService('fiware', 2, null, function(query) {
...
   const endpoint = config.rootURL + '/oauth2/token';
 
+  // Sets dynamic header with clientId and Secret
+  const authHeader = toBase64(`${config.clientId}:${config.secret}`)
```
Comment (mauriciovieira) [inline fiware_server.js]: This code is repeated in a few places. Could you extract `${config.clientId}:${config.secret}` to a function?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "apinf-fiware",
+  "version": "0.0.0",
```
Comment (mauriciovieira) [inline package.json]: Let's release 0.1.0

File: tests/fiware_client.test.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  .get(
+    loginMock.path,
+    loginMock.requestParams
+  )
+  .reply(
+    302,
+    loginMock.responseBody
+  )
+
+describe('requestCredential function', function() {
```
Comment (mauriciovieira) [inline tests/fiware_client.test.js]: describe('requestCredential')

should be a function

File: tests/fiware_server.test.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+      // Catchs thrown error and sets it to error variable
+      err = e
+    }
+
+    // Test done
+    done(err)
+  })
+
+})
```
Comment (mauriciovieira) [inline tests/fiware_server.test.js:77]: Add spacing

File: tests/fiware_server.test.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+    } catch(e) {
+
+      // Catchs thrown error and sets it to error variable
+      err = e
+    }
+
+    // Test done
+    done(err)
+  })
+
```
Comment (mauriciovieira) [inline tests/fiware_server.test.js]: Remove spacing

**Review by mauriciovieira — CHANGES_REQUESTED** [review]
Please do a rebase and fix the following suggestions

File: fiware_server.js (hunk 3 above)
Comment (mauriciovieira) [reply fiware_server.js]: @midyan nevermind, it's going to be done by #16

**Comment by midyan** [comment]
@mauriciovieira 
I have successfully stubbed the popup message. The solution was to simply render and HTML that closes itself, since the access token was already being mocked.

I will fix all the issues your review pointed out and make the code look nicer.

File: fiware_server.js (hunk 2 above)
Comment (mauriciovieira) [reply fiware_server.js]: @midyan fix typo

File: tests/fiware_server.test.js [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+})
+describe('whitelistedFields property', function() {
+  it('should be an array', function(done) {
+    // Error variable
+    let err = null
+
+    // Try/Catch statement throws error if Fiware.whitelistedFields is an array
+    try {
+      // Expects to be array. If not, throw error
+      expect(Fiware.whitelistedFields).to.be.an('array')
```
Comment (mauriciovieira) [inline tests/fiware_server.test.js]: Replace _white listed_ by *allowed* in all places

**Review by mauriciovieira — CHANGES_REQUESTED** [review]
Fix typos and writting, resolve conflicts and merge :-)

File: tests/fiware_server.test.js (hunk 7 above)
Comment (midyan) [reply tests/fiware_server.test.js]: Fixed [here](https://github.com/apinf/apinf-fiware/pull/7/commits/55b81aa50805c513e0c72ab71215054b971a4e9d)

File: tests/fiware_server.test.js (hunk 6 above)
Comment (midyan) [reply tests/fiware_server.test.js:77]: Fixed [here](https://github.com/apinf/apinf-fiware/pull/7/commits/55b81aa50805c513e0c72ab71215054b971a4e9d)

File: fiware_server.js (hunk 2 above)
Comment (midyan) [reply fiware_server.js]: Fixed [here](https://github.com/apinf/apinf-fiware/pull/7/commits/b539bbf24add020f9f367456cde087c3e41d20bd)

File: .travis.yml (hunk 1 above)
Comment (midyan) [reply .travis.yml:10]: Fixed

**Comment by midyan** [comment]
All done :rocket:

**Pull request merged**